### PR TITLE
tests: Fix check for unsupported RoCEv2 devices

### PR DESCRIPTION
**Pull request by Kamalheib**
Avoid the following exception when running RDMATestCase tests over none
Mellanox devices:

Traceback (most recent call last):
  File "/home/rdma-core/tests/test_cq_events.py", line 26, in setUp
    super().setUp()
  File "/home/rdma-core/tests/base.py", line 129, in setUp
    self._add_gids_per_device(ctx, dev_name)
  File "/home/rdma-core/tests/base.py", line 161, in _add_gids_per_device
    self._add_gids_per_port(ctx, dev, port+1)
  File "/home/rdma-core/tests/base.py", line 154, in _add_gids_per_port
    vendor_pid in ROCEV2_UNSUPPORTED_DEVS[dev_attrs.vendor_id]:
KeyError: 5348

Fixes: f65543c15ea4 ("tests: Avoid RoCEv2 GIDs on unsupported devices")
Signed-off-by: Kamal Heib <kamalheib1@gmail.com>

### CODE REVIEW

**Comment by jgunthorpe** [comment]
Why on earth do we need this anyhow? @EdwardSro ? Devices that do not support rocve2 should not be creating v2 gids. Is this because there is no way to find the gid type?

**Comment by EdwardSro** [comment]
> Why on earth do we need this anyhow? @EdwardSro ? Devices that do not support rocve2 should not be creating v2 gids. Is this because there is no way to find the gid type?

We can have a gid of rocev2 type, but we can't know whether the device supports rocev2 or not.
At least that's the case with the device that was added to the "ROCEV2_UNSUPPORTED_DEVS" list, and any other device with the same behavior should be added there. Unless you have a better suggestion.

**Comment by jgunthorpe** [comment]
@EdwardSro oh this is that HW bug in older chips, it should be made clearer, this is not some general mechanism, it is blacklist for a single broken silicon.

**Comment by EdwardSro** [comment]
Yup, it's due to that HW bug. I didn't want to make some comments in Pyverbs that tell stories related to some specific HW bugs and wanted to make the fix as general as possible. But maybe I should have made it more clear in the related commit message.

@Kamalheib Thanks for the fix :)
btw, you can use ".get(key, [])" which will return an empty array in case the key was not found.

**Comment by jgunthorpe** [comment]
You should write it as 'def has_hw_bug' so people can understand what this is and not think it is some expected behavior from verbs.

**Comment by Kamalheib** [comment]
@jgunthorpe & @EdwardSro please review the latest change.

File: tests/base.py [hunk 1]
```diff
@@ -38,6 +38,8 @@
 ROCEV2_UNSUPPORTED_DEVS = {MLNX_VENDOR_ID: [CX3Pro_MLNX_PART_ID,
                                             CX3_MLNX_PART_ID]}
```
Comment (EdwardSro) [inline tests/base.py]: please surround top-level functions and class definitions with two blank lines (PEP conventions)

File: tests/base.py [hunk 2]
```diff
@@ -38,6 +38,8 @@
...
 
+def has_hw_bug(vendor_id, vendor_part_id):
```
Comment (EdwardSro) [inline tests/base.py]: The function name is too generic, let's change it to something like "has_roce_hw_bug"

Comment (rleon) [reply tests/base.py]: maybe has_roce_version_hw_bug() ?

File: tests/base.py (hunk 1 above)
Comment (Kamalheib) [reply tests/base.py]: OK, I will fix it.

File: tests/base.py (hunk 2 above)
Comment (Kamalheib) [reply tests/base.py]: I think has_roce_hw_bug() makes more sense, I'll update the patch.

**Pull request merged**